tests/src/test_cases/test_colorwheel.c

```diff
@@ -0,0 +1,266 @@
+
+    lv_color_hsv_t hsv_cur = lv_colorwheel_get_hsv(colorwheel);
+
+    lv_event_send(colorwheel, LV_EVENT_KEY, &keyCode);
+
+    lv_color_hsv_t hsv_new = lv_colorwheel_get_hsv(colorwheel);
+
+    TEST_ASSERT_EQUAL(hsv_cur.h + 1, hsv_new.h);
+
+    // Change mode to test increase in saturation
+    lv_colorwheel_set_mode(colorwheel, LV_COLORWHEEL_MODE_SATURATION);
+    hsv_cur = hsv_new;
+
+    keyCode = LV_KEY_UP;
+    lv_event_send(colorwheel, LV_EVENT_KEY, &keyCode);
+
+    hsv_new = lv_colorwheel_get_hsv(colorwheel);
+
+    TEST_ASSERT_EQUAL(hsv_cur.s + 1, hsv_new.s);
+
+    // Change mode to test increase in value
+    lv_colorwheel_set_mode(colorwheel, LV_COLORWHEEL_MODE_VALUE);
+    hsv_cur = hsv_new;
+
+    keyCode = LV_KEY_UP;
+    lv_event_send(colorwheel, LV_EVENT_KEY, &keyCode);
+
+    hsv_new = lv_colorwheel_get_hsv(colorwheel);
+
+    TEST_ASSERT_EQUAL(hsv_cur.v + 1, hsv_new.v);
+}
+
+void test_colorwheel_event_keys_left_and_down_decrement_value_by_one(void)
+{
+    uint32_t keyCode = LV_KEY_LEFT;
+
+    // start with a known color to avoid carry over (i.e. %360 / %100)
+    lv_colorwheel_set_hsv(colorwheel, lv_color_rgb_to_hsv(0, 255, 0));
+
+    lv_color_hsv_t hsv_cur = lv_colorwheel_get_hsv(colorwheel);
+
+    lv_event_send(colorwheel, LV_EVENT_KEY, &keyCode);
+
+    lv_color_hsv_t hsv_new = lv_colorwheel_get_hsv(colorwheel);
+
+    TEST_ASSERT_EQUAL(hsv_cur.h - 1, hsv_new.h);
+
+    // Change mode to test decrease in saturation
+    lv_colorwheel_set_mode(colorwheel, LV_COLORWHEEL_MODE_SATURATION);
+    hsv_cur = hsv_new;
+
+    keyCode = LV_KEY_DOWN;
+    lv_event_send(colorwheel, LV_EVENT_KEY, &keyCode);
+
+    hsv_new = lv_colorwheel_get_hsv(colorwheel);
+
+    TEST_ASSERT_EQUAL(hsv_cur.s - 1, hsv_new.s);
+
+    // Change mode to test decrease in value
+    lv_colorwheel_set_mode(colorwheel, LV_COLORWHEEL_MODE_VALUE);
+    hsv_cur = hsv_new;
+
+    keyCode = LV_KEY_DOWN;
+    lv_event_send(colorwheel, LV_EVENT_KEY, &keyCode);
+
+    hsv_new = lv_colorwheel_get_hsv(colorwheel);
+
+    TEST_ASSERT_EQUAL(hsv_cur.v - 1, hsv_new.v);
+}
+
+void test_colorwheel_set_mode_fixed_true(void)
+{
+    lv_colorwheel_set_mode_fixed(colorwheel, true);
+
+    TEST_ASSERT_TRUE(lv_colorwheel_get_color_mode_fixed(colorwheel));
+}
+
+void test_colorwheel_set_mode_fixed_false(void)
+{
+    lv_colorwheel_set_mode_fixed(colorwheel, false);
+
+    TEST_ASSERT_FALSE(lv_colorwheel_get_color_mode_fixed(colorwheel));
+}
+
+void test_colorwheel_angle_to_mode_color_fast_in_mode_value(void)
+{
+    lv_colorwheel_set_mode(colorwheel, LV_COLORWHEEL_MODE_VALUE);
+
+    lv_colorwheel_set_hsv(colorwheel, lv_color_rgb_to_hsv(63, 63, 63));
+
+    TEST_ASSERT_EQUAL_SCREENSHOT("colorwheel_02.png");
+}
+
+void test_colorwheel_angle_to_mode_color_fast_in_mode_saturation(void)
+{
+    lv_colorwheel_set_mode(colorwheel, LV_COLORWHEEL_MODE_SATURATION);
+
+    lv_colorwheel_set_hsv(colorwheel, lv_color_rgb_to_hsv(63, 63, 63));
```

Comment:
Could you use a little bit less black only settings? :slightly_smiling_face:

Reply:
Please find new commit with updated color. Thanks.